pvfactors/engine.py

```diff
@@ -137,32 +145,65 @@ def run_timestep(self, idx):
             pvarray.cast_shadows()
             pvarray.cuts_for_pvrow_view()
 
-            # Calculate view factors
+            # Prepare inputs
             geom_dict = pvarray.dict_surfaces
-            vf_matrix = self.vf_calculator.get_vf_matrix(
-                geom_dict, pvarray.view_matrix, pvarray.obstr_matrix,
-                pvarray.pvrows)
-            pvarray.vf_matrix = vf_matrix
 
             # Apply irradiance terms to pvarray
-            irradiance_vec, invrho_vec = \
+            irradiance_vec, rho_vec, invrho_vec, total_perez_vec = \
                 self.irradiance.transform(pvarray, idx=idx)
 
-            # Calculate radiosities
-            invrho_mat = np.diag(invrho_vec)
-            a_mat = invrho_mat - vf_matrix
-            q0 = linalg.solve(a_mat, irradiance_vec)
-            qinc = np.dot(invrho_mat, q0)
-
-            # Calculate other terms
-            isotropic_vec = vf_matrix[:-1, -1] * irradiance_vec[-1]
-            reflection_vec = qinc[:-1] - irradiance_vec[:-1] - isotropic_vec
-
-            # Update surfaces with values
-            for idx, surface in geom_dict.items():
-                surface.update_params({'q0': q0[idx], 'qinc': qinc[idx],
-                                       'isotropic': isotropic_vec[idx],
-                                       'reflection': reflection_vec[idx]})
+            if self.is_fast_mode:
+                # Indices of the surfaces of the back of the selected pvrows
+                list_surface_indices = pvarray.pvrows[
+                    self.fast_mode_pvrow_index].back.surface_indices
+
+                # TODO: can cut calculation time using a subset of view_matrix
```

Comment:
Is there a big gain doing it?

Reply:
that's virtually where most of the time gain comes from here I believe! I forgot to remove the "TODO" part here sorry, since that's what I actually did